### PR TITLE
Improve software license management #293

### PR DESCRIPTION
**Pull request by wawax**
Improve license management : 
- add a search engine for licenses
- add a new right
- add a new menu entry in Management
- ability to create a license without being in a software form

### CODE REVIEW

File: inc/softwarelicense.class.php [hunk 1]
```diff
@@ -255,11 +255,23 @@ function showForm($ID, $options=array()) {
       echo "<td>";
       if ($ID > 0) {
          $softwares_id = $this->fields["softwares_id"];
+         echo "<a href='software.form.php?id=".$softwares_id."'>".
+                Dropdown::getDropdownName("glpi_softwares", $softwares_id)."</a>";
       } else {
-         echo "<input type='hidden' name='softwares_id' value='$softwares_id'>";
+         //echo "<input type='hidden' name='softwares_id' value='$softwares_id'>";
+         Toolbox::logDebug(array('condition'   => "`is_template`='0' AND `is_deleted`='0'",
```
Comment (orthagh) [inline inc/softwarelicense.class.php]: logdebug restant :)

**Comment by wawax** [comment]
Here is it!

**Comment by prybak** [comment]
You might considering adding features from this plugin - http://plugins.glpi-project.org/#/plugin/fpsoftware.

**Comment by wawax** [comment]
Hi prybak,

I've checked the fpsoftware plugin. I can add a user field in the license form, but I don't understand in which case having several users for a license if useful. Any clue ?

Thanks in advance,

**Comment by KallistiMan** [comment]
You might want to track who and how many users are using a site or enterprise license, to see if it is value for money.

**Comment by prybak** [comment]
Just like stated above you can track how many users are using a site or enterprise licenses.
It is also useful when you track Microsoft software. CALs (SharePoint, Exchange etc.) aren't physical assets, they do not have any keys etc. associated to them. The only thig that represent CAL is your invoice. And as a MS software admin you are responsible for tracking how many users use software like ShP, Exchange.

In general it is useful in any kind of volumne agreements (MS or others) where you count users not 
keys.

It is alos helpful when you need to control access to any cloud service.

**Pull request closed without merging**